File: Algorithms/C/smallest_element.c
#include <stdio.h>

void smallest_element(int* arr, int len, int* smallest, int* index) {
    *smallest = arr[0];
    *index = 0;
    for (int i = 1; i < len; i++) {
        if (arr[i] < *smallest) {
            *smallest = arr[i];
            *index = i;
        }
    }
}

int main() {
    int arr[] = {1, 2, 3, 4, 5, 6};
    int len = sizeof(arr) / sizeof(arr[0]);
    int smallest, index;

    smallest_element(arr, len, &smallest, &index);
    printf("Smallest element is %d at index %d\n", smallest, index);

    return 0;
}

File: Algorithms/Python/smallest_element.py
from __future__ import annotations


def smallest_element(arr: list[int]) -> int:
    """
    Find the smallest element in the array.
    """
    minimum = arr[0]
    index = -1
    for idx, number in enumerate(arr):
        if number < minimum:
            minimum = number
            index = idx
    return index


if __name__ == "__main__":
    list_of_numbers = [12, 3, 4, 5, 6, 7, 8, 9, 10]
    print(smallest_element(list_of_numbers))  # 1
